src/components/MyData/SleepChart/DataSelector.tsx

```diff
@@ -0,0 +1,201 @@
+
+export const DataSelector = (props: Props) => {
+  const { selectionDelayMs = 500 } = props;
+  const { onBlockScrollChange, prepareTooltipData } = props;
+  const common = useCommonChartProps();
+  const [touchStarted, setTouchStarted] = useState<Date>();
+  const [pendingSelection, setPendingSelection] = useState<NativeTouchEvent>();
+  const [selection, setSelection] = useState<NativeTouchEvent>();
+  const { styles } = useStyles(defaultStyles);
+
+  useEffect(() => {
+    let id: number;
+    if (pendingSelection) {
+      id = setTimeout(() => {
+        onBlockScrollChange(true);
+        setSelection(pendingSelection);
+      }, selectionDelayMs) as any as number;
+    }
+
+    return () => clearTimeout(id);
+  }, [pendingSelection, onBlockScrollChange, selectionDelayMs]);
+
+  const clearSelection = useCallback(() => {
+    onBlockScrollChange(false);
+    setSelection(undefined);
+    setPendingSelection(undefined);
+  }, [onBlockScrollChange]);
+
+  const tooltip = useMemo(() => {
+    if (!selection) {
+      return;
+    }
+    const minX = common.padding.left;
+    const maxX = common.plotAreaWidth;
+
+    const tooltipData = {
+      x: selection.locationX,
+      y: selection.locationY,
+      ...prepareTooltipData(Math.min(Math.max(selection.locationX, 0), maxX)),
+    };
+
+    const x = minX + Math.min(Math.max(tooltipData.x, 0), maxX);
+
+    return {
+      ...tooltipData,
+      x,
+      y: common.padding.top,
+      height: common.height - common.padding.bottom - common.padding.top,
+      tooltipX: Math.min(
+        Math.max(x - tooltipOffset, 0),
+        maxX - tooltipWidth / 2,
+      ),
+      displayTooltip: !!tooltipData && !tooltipData.hideTooltip,
+    };
+  }, [selection, prepareTooltipData, common]);
+
+  return (
+    <>
+      {tooltip && (
+        <Svg
+          width={common.width}
+          height={common.height}
+          style={styles.selectionContainer}
+        >
+          <G y={tooltip.y} x={tooltip.x}>
+            <Rect
+              x={-Number(styles.selectionLine?.width ?? 0) / 2}
+              height={tooltip.height}
+              {...styles.selectionLine}
+            />
+          </G>
+          {tooltip.displayTooltip && (
+            <G x={tooltip.tooltipX} y={tooltip.y}>
+              <Rect
+                width={tooltipWidth}
+                height={62}
+                rx={6}
+                {...styles.tooltip}
+              />
+              <Text x={8} y={17} {...styles.tooltipHeader}>
+                {tooltip.header}
+              </Text>
+              <G y={37} x={8}>
+                <Circle x={5} y={-5} r={5} fill={tooltip.color} />
+                <Text {...styles.tooltipTitle} x={16} fontSize={16}>
+                  {tooltip.title}
+                </Text>
+                <Text {...styles.tooltipSubtitle} x={16} y={16}>
+                  {tooltip.subtitle}
+                </Text>
+              </G>
+            </G>
+          )}
+        </Svg>
+      )}
```

Comment:
Lot of hard-coded numbers in here for x and y.  I suppose they're all relative to the parent `<G x={tooltip.tooltipX} y={tooltip.y}>`, and that's how/why we don't want those to be dynamic?